File: src/app/mapper/ldap-user.mapper.ts
import {LdapUser} from "../model/ldap-user.model";

export class LdapUserMapper {
    static toModel(dto: any): LdapUser {
        return {
            dn: dto.dn,
            uid: dto.uid,
            mail: dto.mail
        }
    }
}

File: src/main/kotlin/org/frisp/oss/mailmin/mapper/LdapUserMapper.kt
package org.frisp.oss.mailmin.mapper

import org.frisp.oss.mailmin.dto.LdapUserDTO
import org.frisp.oss.mailmin.ldap.model.LdapUser
import org.springframework.stereotype.Component

@Component
class LdapUserMapper {
    fun toDTO(model: LdapUser): LdapUserDTO {
        return LdapUserDTO(
                dn = model.dn.toString(),
                uid = model.uid,
                mail = model.mail
        )
    }
}
